### ISSUE TITLE
Improve support for huge migration files

### ISSUE THREAD

**Issue by axelfontaine** [flyway/flyway#2259]
We are currently looking to improve support for huge migrations files in Flyway. These files range from 10 MB to several GB or more per file. For this we are looking for your feedback to better understand your usage scenarios. How big are they in your case? What type of statement do these files contain (ex. reference data inserts)? How long is the average statement? How long is the longest statement?

**Comment by rglidden** [comment]
We have three migrations that have been problematic using Flyway SQL migrations due to overall performance. Two of them are reference data inserts, one of which includes a VARBINARY column containing thumbnail images. The other is just a lot of rows of textual and numeric data. The last is a migration that adds a SQL CLR assembly, which also presents itself as a binary blob.

- Max File Size: 5 MB
- Total Lines in Largest File: 12000 Lines
- Average Statement Length: 1000 Characters
- Longest Statement Length: 35000 Characters VARBINARY, 2000 Text

Aside from those reference data inserts, our second largest script (which works fine in 5.2.4) is our baseline, which is only about 8000 lines, 256 KB total size, average length of about 100 characters, and max length of around 1000 - 2000 characters.

The new parser in 6.0.0-Beta seems to make a huge improvement in our case, but due to various breaking changes (that I'm slowly working on reporting), I haven't been able to successfully deploy using 6.0.0-Beta yet.

**Comment by axelfontaine** [comment]
@rglidden Is everything now working correctly for you with 6.0.0-beta2 ?

**Comment by rglidden** [comment]
@axelfontaine Sorry, just getting around to testing it now, but I'm still not able to get a clean run on 6.0.0 Beta 2 on our database due to some additional regression issues compared to 5.2.4 that are causing failures when I try to migrate.  I'll try to do some more testing this week and open issues as I find them.

**Comment by rglidden** [comment]
@axelfontaine With your recent fixes, I was able to get a clean run as of 044b95e1b467e04688b62aeeefb102e7f78519c7. Definitely a big improvement vs. 5.2.4! I found on average the actual execution time on a per-migration basis is slightly slower in 6.0.0, but the big improvement is the pre-migration parsing step.

In our worst case (the VARBINARY reference data inserts) the overall execution time (including parsing) went from 3 minutes in 5.2.4 down to 31 seconds in 6.0.0 despite the actual execution time of the migrations increasing slightly from 15 seconds in 5.2.4 to 20 seconds in 6.0.0.

Our other database with the textual reference data isn't quite as dramatic, but went from 57 seconds total execution time (including parsing) in 5.2.4 down to 37 seconds (including parsing) in 6.0.0.
